server/src/test/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapperTests.java

```diff
@@ -8,93 +8,241 @@
-                .endObject()
-                .endObject()
+        DocumentParsingException e = expectThrows(
+            DocumentParsingException.class,
+            () -> mapper.parse(source(b -> b.startObject("field").field("foo", Arrays.asList(10, 20)).endObject()))
+        );
+        assertEquals(
+            "[sparse_vector] fields take hashes that map a feature to a strictly positive float, but got unexpected token " + "START_ARRAY",
+            e.getCause().getMessage()
         );
 
-        DocumentMapper mapper = mapperService.merge(
-            MapperService.SINGLE_MAPPING_NAME,
-            new CompressedXContent(mapping),
-            MapperService.MergeReason.MAPPING_UPDATE
+        e = expectThrows(DocumentParsingException.class, () -> mapper.parse(source(b -> {
+            b.startArray("foo");
+            {
+                b.startObject().startObject("field").field("bar", 10).endObject().endObject();
+                b.startObject().startObject("field").field("bar", 20).endObject().endObject();
+            }
+            b.endArray();
+        })));
+        assertEquals(
+            "[sparse_vector] fields do not support indexing multiple values for the same rank feature [foo.field.bar] in "
+                + "the same document",
+            e.getCause().getMessage()
         );
+    }
+
+    public void testCannotBeUsedInMultifields() {
+        Exception e = expectThrows(MapperParsingException.class, () -> createMapperService(fieldMapping(b -> {
+            b.field("type", "keyword");
+            b.startObject("fields");
+            b.startObject("feature");
+            b.field("type", "sparse_vector");
+            b.endObject();
+            b.endObject();
+        })));
+        assertThat(e.getMessage(), containsString("Field [feature] of type [sparse_vector] can't be used in multifields"));
+    }
+
+    @Override
+    protected Object generateRandomInputValue(MappedFieldType ft) {
+        assumeFalse("Test implemented in a follow up", true);
+        return null;
+    }
+
+    @Override
+    protected boolean allowsNullValues() {
+        return false;       // TODO should this allow null values?
+    }
+
+    @Override
+    protected SyntheticSourceSupport syntheticSourceSupport(boolean syntheticSource) {
+        throw new AssumptionViolatedException("not supported");
+    }
+
+    @Override
+    protected IngestScriptSupport ingestScriptSupport() {
+        throw new AssumptionViolatedException("not supported");
+    }
+
+    @Override
+    protected String[] getParseMinimalWarnings(IndexVersion indexVersion) {
+        String[] additionalWarnings = null;
+        if (indexVersion.before(IndexVersion.V_8_0_0)) {
+            additionalWarnings = new String[] { SparseVectorFieldMapper.ERROR_MESSAGE_7X };
+        }
+        return Strings.concatStringArrays(super.getParseMinimalWarnings(indexVersion), additionalWarnings);
+    }
+
+    public void testSparseVectorWith7xIndex() throws Exception {
```

Comment:
Added from the previous version of the test. The main idea is to allow indices to exist with the previous `sparse_vector` implementation (<8.0), but no new docs can be added to it.